test/unit/services/instanceAdapter_spec.js

```diff
@@ -0,0 +1,110 @@
+"use strict";
+
+var InstanceAdapter = require("../../../lib/services/instanceAdapter");
+var _               = require("lodash");
+var expect          = require("chai").expect;
+var Genesis         = require("genesis");
+var Bluebird        = require("bluebird");
+var Sinon           = require("sinon");
+require("sinon-as-promised")(Bluebird);
+
+describe("The Adapter class ", function () {
+	it("is frozen", function () {
+		expect(Object.isFrozen(InstanceAdapter), "frozen").to.be.true;
+	});
+
+	function describeCreate (description, options, fails) {
+		var optionsToPass = _.clone(options, true);
+		var expectedOptions = _.defaults(
+			options,
+			{
+				id    : undefined,
+				type  : "t2.micro",
+				ami   : "ami-d05e75b8",
+				state : "pending",
+				uri   : null
+			}
+		);
+
+		describe("creating a new instance " + description, function () {
+			var instances = new InstanceAdapter();
+
+			if (!fails) {
+				it("returns a new instance", function () {
+					var result;
+					instances.createInstance(optionsToPass.ami, optionsToPass.type)
+					.then(function (model) {
+						result = model;
```

Comment:
Does this mean you want me to check if result is of type instance and not whether the values are equal to the expected values because that's been tested while testing the Instance class?